metricbeat/module/system/test_system.py

```diff
@@ -430,11 +434,13 @@ def test_process(self):
             process.pop("cgroup", None)
             process.pop("fd", None)
             process.pop("cmdline", None)
+            process.pop("num_threads", None)
 
             self.assertCountEqual(SYSTEM_PROCESS_FIELDS, process.keys())
-
-            self.assertTrue(
-                found_cmdline, "cmdline not found in any process events")
+        # After iterating over all process, make sure at least one of them had
+        # the 'cmdline' set.
+        self.assertTrue(
+            found_cmdline, "cmdline not found in any process events")
```

Comment:
@fearful-symmetry, could you do a sanity check on this last change? I believe it is ok to change, but I'd like a second opinion before merging.

Reply:
It looks like the `assertTrue()` line was just moved out of the loop block to check for the cmdline globally? That seems fine, but wouldn't it require the last event to set `found_cmdline |= "cmdline" in process` in order to pass? Maybe a counter that needs to be > 1 would work?

Reply:
> but wouldn't it require the last event to set found_cmdline |= "cmdline" in process in order to pass?

Reply:
> It looks like the assertTrue() line was just moved out of the loop block to check for the cmdline globally?

Yes, that was my intent because it already does an `OR` operation on itself. Being honest I was very confused by the `assertTrue()` being inside the loop and having `found_cmdline |= "cmdline" in process`.

https://github.com/elastic/beats/blob/bab5bb460c985cf0fa129759e3c891bf7f9ff65d/metricbeat/module/system/test_system.py#L421-L427

> but wouldn't it require the last event to set found_cmdline |= "cmdline" in process in order to pass?

That's an `OR` operator, so as long as `"cmdline" in process` returns `True` at least once, then `found_cmdline` will always be `True`.